thoth/messaging/unresolved_package.py

```diff
@@ -36,7 +36,7 @@ class MessageContents(BaseMessageContents):
 
     package_name: StrictStr
     package_version: Optional[StrictStr]
-    index_url: Optional[StrictStr]
+    index_url: Optional[List[StrictStr]]
```

Comment:
Why is a list propagated here?

Reply:
you might be right, index has always just one URL at any time, right?

Reply:
I'm going base off what is expected in `investigator`[0] and sent by `graph-refresh-job`[1]

I  can change both of these instances send a single index url instead of the change seen here.

[1] https://github.com/thoth-station/investigator/blob/bff0c51e4c106a0755e06e7800918c71d0fa5fdc/thoth/investigator/unresolved_package/investigate_unresolved_package.py#L51
[2] https://github.com/thoth-station/graph-refresh-job/blob/d61d8764238af2fc8846cdce86fc06967cfe7715/producer.py#L205

Reply:
Yes, please keep just one index per message. It will simplify consumers logic and any possible integrations+monitoring.

Reply:
Right now if no solver is passed all solvers are schedule. Should this also change so that it is on the consumer to send messages for all registered solvers?

Reply:
From design point of view, it could be good to have this on producer side so we have the whole triplet (package_name, package_version, index_url) in the message stated.